lib/kb_Metrics/metricsdb_controller.py

```diff
@@ -26,279 +26,189 @@ def _config_str_to_list(self, list_str):
-    def get_active_users_counts(self, requesting_user, params, token, exclude_kbstaff=True):
-        if not self.is_metrics_admin(requesting_user):
-            raise ValueError('You do not have permission to view this data.')
-
-        params = self.process_parameters(params)
-
-        if exclude_kbstaff:
-            mt_ret = self.metrics_dbi.aggr_unique_users_per_day(params['minTime'],
-                                                                params['maxTime'],
-                                                                self.kbstaffList)
-        else:
-            mt_ret = self.metrics_dbi.aggr_unique_users_per_day(
-                params['minTime'], params['maxTime'], [])
-
-        if len(mt_ret) == 0:
-            print("No records returned!")
-
-        return {'metrics_result': mt_ret}
-
-    def get_user_details(self, requesting_user, params, token, exclude_kbstaff=False):
-        if not self.is_metrics_admin(requesting_user):
-            raise ValueError('You do not have permission to view this data.')
-
-        params = self.process_parameters(params)
-        mt_ret = self.metrics_dbi.get_user_info(params['user_ids'], params['minTime'],
-                                                params['maxTime'], exclude_kbstaff)
-        if len(mt_ret) == 0:
-            print("No records returned!")
-        else:
-            mt_ret = self.convert_isodate_to_millis(mt_ret,
-                                                    ['signup_at', 'last_signin_at'])
-        return {'metrics_result': mt_ret}
-
-    def get_activities(self, requesting_user, params, token):
-        # TODO not yet pointing to the metrics db yet
-        if not self.is_metrics_admin(requesting_user):
-            raise ValueError('You do not have permission to view this data.')
-        return self.get_activities_from_wsobjs(requesting_user, params, token)
-
-    def get_narratives(self, requesting_user, params, token):
-        # TODO not yet pointing to the metrics db yet
-        if not self.is_metrics_admin(requesting_user):
-            raise ValueError('You do not have permission to view this data.')
-        return self.get_narratives_from_wsobjs(requesting_user, params, token)
-
-    # End functions to get the requested records from metrics db
-
     # functions to get the requested records from other dbs...
-    def get_narratives_from_wsobjs(self, requesting_user, params, token):
-        if not self.is_admin(requesting_user):
+    def _get_narratives_from_wsobjs(self, requesting_user, params, token):
+        """
+        get_narratives_from_wsobjs--Given a time period, fetch the narrative
+        information from workspace.workspaces and workspace.workspaceObjects.
+        Based on the narratives in workspace.workspaceObjects, if additional
+        info available then add to existing data from workspace.workspaces.
+        """
+        if not self._is_metrics_admin(requesting_user):
```

Comment:
Same for _get_narratives_from_wsobjs. _is_metrics_admin is already checked in upstream funcs.

Reply:
Same answer as previous.